File: app/src/main/java/com/lucas/omnia/utils/SharedPreferencesHelper.java
package com.lucas.omnia.utils;

import android.content.SharedPreferences;

/**
 *  Helper class to manage access to {@link SharedPreferences}.
 */
public class SharedPreferencesHelper {

    // Keys for saving values in SharedPreferences.
    public static final String KEY_ID = "key_id";
    public static final String KEY_USERNAME = "key_username";

    // The injected SharedPreferences implementation to use for persistence.
    private final SharedPreferences mSharedPreferences;

    /**
     * Constructor with dependency injection.
     *
     * @param sharedPreferences The {@link SharedPreferences} that will be used in this DAO.
     */
    public SharedPreferencesHelper(SharedPreferences sharedPreferences) {
        mSharedPreferences = sharedPreferences;
    }

    /**
     * Saves the given {@link SharedPreferenceEntry} that contains the user's settings to
     * {@link SharedPreferences}.
     *
     * @param sharedPreferenceEntry contains data to save to {@link SharedPreferences}.
     * @return {@code true} if writing to {@link SharedPreferences} succeeded. {@code false}
     *         otherwise.
     */
    public boolean saveUserInfo(SharedPreferenceEntry sharedPreferenceEntry){
        // Start a SharedPreferences transaction.
        SharedPreferences.Editor editor = mSharedPreferences.edit();
        editor.putString(KEY_ID, sharedPreferenceEntry.getUid());
        editor.putString(KEY_USERNAME, sharedPreferenceEntry.getUsername());

        // Commit changes to SharedPreferences.
        return editor.commit();
    }

    /**
     * Retrieves the {@link SharedPreferenceEntry} containing the user's personal information from
     * {@link SharedPreferences}.
     *
     * @return the Retrieved {@link SharedPreferenceEntry}.
     */
    public SharedPreferenceEntry getUserInfo() {
        // Get data from the SharedPreferences.
        String uid = mSharedPreferences.getString(KEY_ID, "");
        String username = mSharedPreferences.getString(KEY_USERNAME, "");

        // Create and fill a SharedPreferenceEntry model object.
        return new SharedPreferenceEntry(uid, username);
    }
}

File: app/src/main/java/com/lucas/omnia/utils/SharedPreferencesHelper.kt
package com.lucas.omnia.utils

import android.content.SharedPreferences

/**
 * Helper class to manage access to [SharedPreferences].
 *
 * @param sharedPreferences The injected [SharedPreferences] that will be used in this DAO.
 */
class SharedPreferencesHelper(private val sharedPreferences: SharedPreferences) {

    /**
     * Retrieves the [SharedPreferenceEntry] containing the user's personal information from
     * [SharedPreferences].
     *
     * @return the Retrieved [SharedPreferenceEntry].
     */
    // Get data from the SharedPreferences.
    // Create and fill a SharedPreferenceEntry model object.
    fun getPersonalInfo(): SharedPreferenceEntry {
        val uid = sharedPreferences.getString(KEY_ID, "")
        val username = sharedPreferences.getString(KEY_USERNAME, "")
        return SharedPreferenceEntry(uid, username)
    }

    /**
     * Saves the given [SharedPreferenceEntry] that contains the user's settings to
     * [SharedPreferences].
     *
     * @param sharedPreferenceEntry contains data to save to [SharedPreferences].
     * @return `true` if writing to [SharedPreferences] succeeded, `false` otherwise.
     */
    fun savePersonalInfo(sharedPreferenceEntry: SharedPreferenceEntry): Boolean {
        // Start a SharedPreferences transaction.
        val editor = sharedPreferences.edit().apply() {
            putString(KEY_ID, sharedPreferenceEntry.uid)
            putString(KEY_USERNAME, sharedPreferenceEntry.username)
        }

        // Commit changes to SharedPreferences.
        return editor.commit()
    }

    companion object {
        // Keys for saving values in SharedPreferences.
        const val KEY_ID = "key_id"
        const val KEY_USERNAME = "key_username"
    }
}